docs/how-to/Bar-Memory.rst

```diff
@@ -0,0 +1,102 @@
+Troubleshoot BAR access limitation
+**************************************
+A system can consist of multiple PCIe devices and data access between the devices is needed for proper functioning and optimum system performance. Direct Memory Access (DMA) to the device resources using the BARs (Base Address Registers) can be restricted due to the physical address limit. This restriction can result in data access failure between the system components. Peer-to-peer (P2P) DMA is used to access the resources such as registers and device memory between the devices. These PCIe devices need memory-mapped input/output (MMIO) space for DMA and are defined in the PCIe BARs. These BARs are also used by the CPU and other devices in the system to access the resources of the PCIe devices. P2P DMA only works when one device can directly access the local BAR memory of another. If the BAR memory is above the physical addressing limit of the device, the device will not be able to access the remote BAR. 
+
+To handle the issue with BAR access, you need to be aware of the physical address limitations of the devices and understand the :ref:`BAR configuration of AMD GPUs <bar-configuration>`. This information is important when you need to set up additional MMIO apertures for PCIe devices in the physical address space of the system.
+
+Handling physical address limitation
+=============================================
+When the system boots, the system BIOS allocates the physical address space for different components in the system. This also includes system memory and one or more MMIO apertures. In general, there are two MMIO apertures: one allocated with below 4 GB of physical address space for 32-bit compatibility, and one with above 4 GB for devices that need more space. 
+
+You can control the memory address of the high MMIO aperture from the configuration options in the system BIOS. This enables you to configure the additional MMIO space to align with the physical addressing limit and allow P2P DMA between the devices. For example, if a PCIe device is limited to 44-bit of physical addressing, you should ensure that the MMIO aperture is set below 44-bit in the system physical address space.
+
+There are two ways to handle this:
+
+* Ensure that the high MMIO aperture is within the physical addressing limits of the devices in the system. For example, if the devices have a 44-bit physical addressing limit, set the ``MMIOH Base`` and ``MMIO High size`` options in the BIOS such that the aperture is within the 44-bit address range. Also, ensure that the ``Above 4G Decoding`` option is Enabled. 
+
+* Enable the Input-Output Memory Management Unit (IOMMU). When the IOMMU is enabled in the non-passthrough mode, it will create a virtual IO address space for each device on the system. It also ensures that all virtual addresses created in that space are within the physical addressing limits of the device. For more information on IOMMU, see :doc:`../conceptual/iommu`. 
```

Comment:
```suggestion
* Enable the Input-Output Memory Management Unit (IOMMU). When the IOMMU is enabled in non-passthrough mode, it will create a virtual I/O address space for each device on the system. It also ensures that all virtual addresses created in that space are within the physical addressing limits of the device. For more information on IOMMU, see :doc:`../conceptual/iommu`. 
```